### ISSUE TITLE
Make everything use serialization versions right (removing ONLY_VERSION)

### ISSUE THREAD

**Issue by srh** [rethinkdb/rethinkdb#2563]
In followup to #2537, but less urgent relative to other work on 1.14., we need to make everything use serialization versions properly.  Many things still assume there's just one version.  These things use `cluster_version_t::ONLY_VERSION` to declare this assumption.

For example: when sending a message across the cluster, right now we still use ONLY_VERSION instead of having a map of connection-sessions and version numbers to look that information up by.

For example: persist.cc assumes there's just one version, with no code that checks the block magic.

Out-of-scope example: The btree code assumes there's just one version of leaf format.

Out-of-scope example: When upgrading a cluster with changes to the query language between versions, ReQL evaluation can't start using new query language features until... everything supports it.  There's no code, no UI, related to bringing the cluster up a version, or, say, bringing a table up a version, if the per-table notion would make sense.  (See #2564.)

(Assigning to myself, because this is a continuation of #2537.)

(Implementing in sam_2563, branched off of sam_2537, which is branched off of v1.13.x.)
